### PR TITLE
Convert to AWS-SDK and add STS::AssumeRole support

### PR DESCRIPTION
**Pull request by tlunter**
We needed to add STS::AssumeRole support, but unfortunately Fog does not support this form of authentication.

So, first I upgrade the gem to AWS-SDK, then added support for assuming roles, and then fixed all the bugs in the code/tests.

Finally, I readded back the old defaults for launch configurations to make it easier to transition to this version of the gem.

### CODE REVIEW

**Comment by tlunter** [comment]
@adamjt  @bfulton @nahiluhmot

**Comment by adamjt** [comment]
A lot to digest here but looks great. Travis seems up set about it though.

**Comment by tlunter** [comment]
Cane was complaining, should be green after this!

File: lib/aerosol/auto_scaling.rb [hunk 1]
```diff
@@ -155,85 +149,20 @@ def conn
...
-        tag key => value
-      end
-    else
-      ary.each do |hash|
-        tag hash['Key'] => hash['Value']
-      end
-    end
-  end
-
-  # Unfortunately, Fog does not create fake instances after an auto scaling
```
Comment (nahiluhmot) [inline lib/aerosol/auto_scaling.rb]: Can't be happier about this dying.

File: lib/aerosol/launch_configuration.rb [hunk 2]
```diff
@@ -65,21 +69,21 @@ def self.request_all
 
     begin
       new_lcs = request_all_for_token(next_token)
-      lcs.concat(new_lcs['LaunchConfigurations'])
-      next_token = new_lcs['NextToken']
-    end while !next_token.nil?
+      lcs.concat(new_lcs.launch_configurations)
+      next_token = new_lcs.next_token
+    end until next_token.nil?
```
Comment (nahiluhmot) [inline lib/aerosol/launch_configuration.rb]: Ah, the good old `begin...until` loop!

**Comment by nahiluhmot** [comment]
I take it this will be a major version bump?

**Comment by tlunter** [comment]
I think it should since we're changing dependencies.  I did try and minimize any differences between the versions otherwise.  I did have to add a warning for `ami` and `iam_role` though since that's not was AWS calls it in the API.

**Comment by tlunter** [comment]
Anymore questions about this?

**Comment by adamjt** [comment]
I'm +1

**Comment by nahiluhmot** [comment]
:+1:

**Pull request merged**